src/main/java/io/github/syst3ms/skriptparser/util/math/BigDecimalMath.java

```diff
@@ -711,7 +711,7 @@ private static BigDecimal logUsingTwoThree(BigDecimal x, MathContext mathContext
             result = result.subtract(logThree(mcDouble).multiply(valueOf(-factorOfThree), mc), mc);
         }
 
-        if (x == correctedX && result == ZERO) {
+        if (x.equals(correctedX) && result.equals(ZERO)) {
```

Comment:
No.

You don't use `==` on BigDecimals for the same reason that you don't use `==` on Strings.

Reply:
Okay, that's literally what I replaced?

Reply:
Oops lmao, read it backwards.